lib/driver.js

```diff
@@ -54,6 +54,14 @@ module.exports = R.curry((server_url, Client) => {
       Client.put( DEFAULT_HEADERS, BucketUrl(bucket, docid), doc )
     )
 
+    // revid is the latest revision id needed to update an existing document
+  , upsert: R.curry((bucket, docid, revid, doc) => {
```

Comment:
gotcha. so we can expect for update to be in got-couch and then an upsert function can be built on the kuss side.